sdk/core/core/CMakeLists.txt

```diff
@@ -71,28 +71,28 @@ target_include_directories(${TARGET_NAME} PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRE
 # include test internal headers
 target_include_directories(${TARGET_NAME} PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/test/inc>)
 
-set(AZ_PLATFORM_IMPL "NONE" CACHE STRING "Platform implementation to use")
-set_property(CACHE AZ_PLATFORM_IMPL PROPERTY STRINGS NONE WIN32 POSIX USER)
+set(PLATFORM "NONE" CACHE STRING "Platform implementation to use")
+set_property(CACHE PLATFORM PROPERTY STRINGS NONE WIN32 POSIX CUSTOM)
 
-set(AZ_USER_PLATFORM_IMPL_NAME "" CACHE STRING "Name of the implementation target")
+set(CUSTOM_PLATFORM_NAME "" CACHE STRING "Name of the implementation target")
 
-if(AZ_PLATFORM_IMPL STREQUAL "USER")
-  set(AZ_PLATFORM_IMPL_USER ON)
-elseif(AZ_PLATFORM_IMPL STREQUAL "POSIX")
-  set(AZ_PLATFORM_IMPL_POSIX ON)
-elseif(AZ_PLATFORM_IMPL STREQUAL "WIN32")
-  set(AZ_PLATFORM_IMPL_WIN32 ON)
+if(PLATFORM STREQUAL "CUSTOM")
+  set(PLATFORM_CUSTOM ON)
+elseif(PLATFORM STREQUAL "POSIX")
+  set(PLATFORM_POSIX ON)
+elseif(PLATFORM STREQUAL "WIN32")
+  set(PLATFORM_WIN32 ON)
 else()
-  set(AZ_PLATFORM_IMPL_NONE ON)
+  set(PLATFORM_NONE ON)
 endif()
```

Comment:
Since you are already modifying this, can we get rid of `PLATFORM_CUSTOM`, `PLATFORM_POSIX`, etc. and the if blocks below, and just move the  `target_link_libraries` instructions into this if-block? We can simplify the logic here.

For example:
```text
if(PLATFORM STREQUAL "CUSTOM")
   target_link_libraries(${TARGET_NAME} PRIVATE ${CUSTOM_PLATFORM_NAME})
elseif(PLATFORM STREQUAL "POSIX")
   target_link_libraries(${TARGET_NAME} PRIVATE az_posix)
...

It doesn't look like we use the `PLATFORM_*` names anywhere else anyway, so there is no point setting them.
```

Reply:
Thank you, good suggestion! Done.